x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -112,104 +130,130 @@ public ModelLoadingService(TrainedModelProvider trainedModelProvider,
         this.modelStatsService = modelStatsService;
         this.shouldNotAudit = new HashSet<>();
         this.namedXContentRegistry = namedXContentRegistry;
-        this.localModelCache = CacheBuilder.<String, LocalModel>builder()
+        this.localModelCache = CacheBuilder.<String, ModelAndConsumer>builder()
             .setMaximumWeight(this.maxCacheSize.getBytes())
-            .weigher((id, localModel) -> localModel.ramBytesUsed())
+            .weigher((id, modelAndConsumer) -> modelAndConsumer.model.ramBytesUsed())
             // explicit declaration of the listener lambda necessary for Eclipse IDE 4.14
-            .removalListener(notification -> cacheEvictionListener(notification))
+            .removalListener(this::cacheEvictionListener)
             .setExpireAfterAccess(INFERENCE_MODEL_CACHE_TTL.get(settings))
             .build();
         clusterService.addListener(this);
         this.localNode = localNode;
     }
 
+    public boolean modelIsCached(String modelId) {
```

Comment:
```suggestion
    boolean modelIsCached(String modelId) {
```
Lets make this package private so folks aren't tempted to use it. In a real scenario, this could return true, and then the model is removed when the caller attempts to get it.